src/world/FieldMessage.cpp

```diff
@@ -6,214 +6,139 @@
+    for (proto::SSL_FieldCicularArc arc : sslFieldSize.field_arcs()) {
+        FieldArc newArc;
+        if (NAME_MAP.count(arc.name()) > 0) {
+            newArc.name = NAME_MAP[arc.name()];
+            newArc.center = mm_to_m(arc.center());
+            newArc.a1 = mm_to_m(arc.a1());
+            newArc.a2 = mm_to_m(arc.a2());
+            newArc.radius = mm_to_m(arc.radius());
+            newArc.thickness = mm_to_m(arc.thickness());
+            FieldArcName fieldArcName = CONVERT_TO_FIELD_ARC_NAME.at(newArc.name);
+            fieldArcs[fieldArcName] = newArc;
+        }
     }
 }
 
-void FieldMessage::addHelperLine(FieldLineSegment newLine) {
-  if (newLine.name == "top_line") {
-      top_line = newLine;
-    } else if (newLine.name == "bottom_line") {
-      bottom_line = newLine;
-    } else if (newLine.name == "left_line") {
-      left_line = newLine;
-    } else if (newLine.name == "right_line") {
-      right_line = newLine;
-    } else if (newLine.name == "half_line") {
-      half_line = newLine;
-    } else if (newLine.name == "center_line") {
-      center_line = newLine;
-    } else if (newLine.name == "left_penalty_line") {
-      left_penalty_line = newLine;
-    } else if (newLine.name == "right_penalty_line") {
-      right_penalty_line = newLine;
-    }
-      // adding rectangle box lines
-    else if (newLine.name == "top_left_penalty_stretch") {
-      top_left_penalty_stretch = newLine;
-    } else if (newLine.name == "bottom_left_penalty_stretch") {
-      bottom_left_penalty_stretch = newLine;
-    } else if (newLine.name == "top_right_penalty_stretch") {
-      top_right_penalty_stretch = newLine;
-    } else if (newLine.name == "bottom_right_penalty_stretch") {
-      bottom_right_penalty_stretch = newLine;
-    }
+void FieldMessage::initFieldVectors() {
+    fieldVectors[OUR_GOAL_CENTER] = Vector2(fieldValues[LEFTMOST_X], fieldValues[CENTER_Y]);
+    fieldVectors[THEIR_GOAL_CENTER] = Vector2(fieldValues[RIGHTMOST_X], fieldValues[CENTER_Y]);
+
+    Vector2 goalWidthAdjust = Vector2(0, fieldValues[GOAL_WIDTH] / 2);
+    fieldVectors[OUR_BOTTOM_GOAL_SIDE] = fieldVectors[OUR_GOAL_CENTER] - goalWidthAdjust;
+    fieldVectors[OUR_TOP_GOAL_SIDE] = fieldVectors[OUR_GOAL_CENTER] + goalWidthAdjust;
+    fieldVectors[THEIR_BOTTOM_GOAL_SIDE] = fieldVectors[THEIR_GOAL_CENTER] - goalWidthAdjust;
+    fieldVectors[THEIR_TOP_GOAL_SIDE] = fieldVectors[THEIR_GOAL_CENTER] + goalWidthAdjust;
+
+    Vector2 lpl_begin = fieldLines[LEFT_PENALTY_LINE].begin;
+    Vector2 lpl_end = fieldLines[LEFT_PENALTY_LINE].end;
+    fieldVectors[LEFT_PENALTY_POINT] = lpl_begin + ((lpl_end - lpl_begin) * 0.5);
+
+    Vector2 rpl_begin = fieldLines[RIGHT_PENALTY_LINE].begin;
+    Vector2 rpl_end = fieldLines[RIGHT_PENALTY_LINE].end;
+    fieldVectors[RIGHT_PENALTY_POINT] = rpl_begin + ((rpl_end - rpl_begin) * 0.5);
 }
 
 float FieldMessage::mm_to_m(float scalar) {
-  return scalar/1000;
+    return scalar / 1000;
```

Comment:
The function is indeed now changed to static, however I do not agree that the return value should be const. Because it should be possible to multiply it by 5 if you want 5 times the length of something.

Reply:
The return value shouldn't be constant, but the Field message itself should be:
these are 2 different things:

1: const int car::wheelCount();
2: int car::wheelCount() const;

The first returns a const. integer, but it can change the internal state of the car.

The second returns an integer, but it is not allowed to change the internal state of the car (e.g. the car is constant/not modified). I'm suggesting you use the latter.